### PR TITLE
[operator] fix: move 'relabelings' and 'metricRelabelings' configs under endpoints in serviceMonitor spec

### PR DESCRIPTION
**Pull request by rl0nergan**
Fixes indentation issues causing an invalid ServiceMonitor spec when using relabeling/metricRelabeling configs  https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1287

### CODE REVIEW

File: charts/opentelemetry-operator/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -24,12 +24,12 @@ spec:
...
-  {{- toYaml . | nindent 4 }}
+    relabelings:
+  {{- toYaml . | nindent 6 }}
   {{- end }}
   {{- with .Values.manager.serviceMonitor.metricRelabelings }}
-  metricRelabelings:
-  {{- toYaml . | nindent 4 }}
+    metricRelabelings:
+  {{- toYaml . | nindent 6 }}
   {{- end }}
```
Comment (jaronoff97) [inline charts/opentelemetry-operator/templates/servicemonitor.yaml:33]: suggestion: can we have all the blocks indentation match here?

Comment (rl0nergan) [reply charts/opentelemetry-operator/templates/servicemonitor.yaml]: good call. just pushed those changes up

**Review by jaronoff97 — APPROVED** [review]

**Review by Allex1 — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
I merged the operator-bump first, so needs a rebase

**Comment by rl0nergan** [comment]
using the following values for the serviceMonitor

```
  serviceMonitor:
    enabled: true
    # additional labels on the ServiceMonitor
    extraLabels: {}
    # add annotations on the ServiceMonitor
    annotations: {}
    metricsEndpoints:
      - port: metrics
    # Used to set relabeling and metricRelabeling configs on the ServiceMonitor
    # https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config
    relabelings:
      - action: replace
        replacement: my-cluster
        targetLabel: cluster
      - action: replace
        replacement: my-cluster2
        targetLabel: cluster2
    metricRelabelings:
      - action: keep
        regex: otelcol_process_memory_rss|target_info
        sourceLabels:
        - __name__
```

Templates out to this:
```
# Source: opentelemetry-operator/templates/servicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: otel-operator-opentelemetry-operator
  namespace: redpanda
  labels:
    helm.sh/chart: opentelemetry-operator-0.64.5
    app.kubernetes.io/name: opentelemetry-operator
    app.kubernetes.io/version: "0.103.0"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: otel-operator
    
    app.kubernetes.io/component: controller-manager
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: opentelemetry-operator
      app.kubernetes.io/component: controller-manager
  endpoints:
  - port: metrics
    relabelings:
      - action: replace
        replacement: my-cluster
        targetLabel: cluster
      - action: replace
        replacement: my-cluster2
        targetLabel: cluster2
    metricRelabelings:
      - action: keep
        regex: otelcol_process_memory_rss|target_info
        sourceLabels:
        - __name__
  namespaceSelector:
    matchNames:
      - redpanda
```

Ignore the namespacing, that's just coming from the namespace set in my current kubeconfig.

**Comment by rl0nergan** [comment]
Updated PR with additional indentation:
```
# Source: opentelemetry-operator/templates/servicemonitor.yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: otel-operator-opentelemetry-operator
  namespace: redpanda
  labels:
    helm.sh/chart: opentelemetry-operator-0.65.1
    app.kubernetes.io/name: opentelemetry-operator
    app.kubernetes.io/version: "0.104.0"
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/instance: otel-operator
    
    app.kubernetes.io/component: controller-manager
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: opentelemetry-operator
      app.kubernetes.io/component: controller-manager
  endpoints:
    - port: metrics
      relabelings:
        - action: replace
          replacement: my-cluster
          targetLabel: cluster
        - action: replace
          replacement: my-cluster2
          targetLabel: cluster2
      metricRelabelings:
        - action: keep
          regex: otelcol_process_memory_rss|target_info
          sourceLabels:
          - __name__
  namespaceSelector:
    matchNames:
      - redpanda
```

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**